### PR TITLE
refactor(middleware): use v2 endpoints

### LINKED ISSUE #1431: Adjust Middleware interface to V2

### PR DESCRIPTION
**Pull request by CedrikNikita**
fixes #1431

### CODE REVIEW

File: src/store/plugins/names.js [hunk 1]
```diff
@@ -105,19 +105,20 @@ export default (store) => {
...
             getPendingNameClaimTransactions(address),
-            middleware.getNamesOwnedBy(address)
-              .then(({ active }) => active.map(({ info, name, hash }) => ({
-                createdAtHeight: info.activeFrom,
-                expiresAt: info.expireHeight,
-                owner: info.ownership.current,
-                pointers: info.pointers,
-                autoExtend: !!(
+            middleware.getNames({ owned_by: address })
+              .then(({ data }) => data.filter(({ active }) => active)
```
Comment (peronczyk) [inline src/store/plugins/names.js:109]: I suggest to move the `.filter` to the next line to make the code more readable.

File: src/store/plugins/names.js [hunk 2]
```diff
@@ -105,19 +105,20 @@ export default (store) => {
...
+                .map(({ info, name, hash }) => ({
+                  createdAtHeight: info.activeFrom,
+                  expiresAt: info.expireHeight,
+                  owner: info.ownership.current,
+                  pointers: info.pointers,
+                  autoExtend: !!(
```
Comment (peronczyk) [inline src/store/plugins/names.js:115]: Next 2 lines should also be indented.

File: src/composables/middleware.ts
Comment (peronczyk) [inline src/composables/middleware.ts:1]: I suggest to update the `fetchFromMiddleware` to use the `/v2` path by default. Now sometimes running this function requires using the `v2/` and sometimes no.

File: src/composables/middleware.ts [hunk 3]
```diff
@@ -64,21 +64,17 @@ export function useMiddleware({ store }: IDefaultComposableOptions) {
               required: true,
               type: 'integer',
             },
+            {
+              in: 'query',
+              name: 'direction',
+              default: 'backward',
+              type: 'integer',
```
Comment (peronczyk) [inline src/composables/middleware.ts:71]: Are you sure this should be an integer?

**Review by peronczyk — CHANGES_REQUESTED** [review]

File: src/composables/middleware.ts
Comment (CedrikNikita) [reply src/composables/middleware.ts:1]: The `next` property including `/v2` in the route, and at this point we would need to make the logic of fetching all pages more complicated.

**Comment by Liubov-crypto** [comment]
I can't see all my registered chain names in my account. On dev branch I can see them all.

File: src/store/plugins/names.js [hunk 4]
```diff
@@ -105,19 +105,22 @@ export default (store) => {
...
+            middleware.getNames({ owned_by: address })
+              .then(({ data }) => data
+                .filter(({ active }) => active)
+                .map(({ info, name, hash }) => ({
+                  createdAtHeight: info.activeFrom,
+                  expiresAt: info.expireHeight,
+                  owner: info.ownership.current,
+                  pointers: info.pointers,
+                  autoExtend:
+                  !!(
```
Comment (peronczyk) [inline src/store/plugins/names.js:117]: I don't see a reason to move this into new line. I think the way how it was before was OK.

**Review by peronczyk — APPROVED** [review]

**Comment by Liubov-crypto** [comment]
The issue with names has been fixed. LGTM.

**Comment by Liubov-crypto** [comment]
Retested, LGTM

**Review by konradodo — APPROVED** [review]

**Pull request merged**